randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -204,6 +223,33 @@ def game_enum(self) -> RandovaniaGame:
     def hint_namer_type(cls) -> type[MSRHintNamer]:
         return MSRHintNamer
 
+    def get_map_icon_for_resources(
+        self, pickup: PickupEntry, conditional_resources: list[ConditionalResources], is_remote: bool
+    ) -> str:
+        def get_icon_from_item_res(item: ResourceInfo) -> str:
+            assert isinstance(item, ItemResourceInfo)
+            if "icon" in item.extra:
+                return item.extra["icon"]
+            return "item_powerup"
+
+        def get_icon_from_cond_res(cond_res: ConditionalResources) -> str:
+            if not cond_res.resources:
+                return "item_nothing"
+            for resource in cond_res.resources:
+                return get_icon_from_item_res(resource[0])
+            return "item_powerup"
+
+        # TODO: Do something with self.configuration.pickup_model_style
```

Comment:
What styles from all these different methods for hiding things, should show which icon?